File: src/main/java/com/qwwuyu/server/dao/CardMapper.java
package com.qwwuyu.server.dao;

import com.qwwuyu.server.bean.Card;
import org.apache.ibatis.annotations.Param;

import java.util.List;

public interface CardMapper {
    Card selectByPrimaryKey(Integer id);

    int insert(Card record);

    int updateByPrimaryKeySelective(Card record);

    int deleteByPrimaryKey(Integer id);

    List<Card> selectByCard(@Param("card") Card card, @Param("limit") int limit, @Param("offset") int offset, @Param("asc") String asc, @Param("desc") String desc);
}

File: app/src/main/kotlin/com/qwwuyu/gs/mapper/CardMapper.kt
package com.qwwuyu.gs.mapper

import com.qwwuyu.gs.entity.Card
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Param

@Mapper
interface CardMapper {
    fun selectByPrimaryKey(id: Int): Card?

    fun insert(record: Card): Int

    fun updateByPrimaryKeySelective(record: Card): Int

    fun deleteByPrimaryKey(id: Int): Int

    fun selectByCard(@Param("card") card: Card, @Param("limit") limit: Int, @Param("offset") offset: Int,
                     @Param("asc") asc: String?, @Param("desc") desc: String?): List<Card>
}